static/js/components/SourcesInGcn.jsx

```diff
@@ -0,0 +1,267 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { useSelector, useDispatch } from "react-redux";
+import Paper from "@mui/material/Paper";
+import { withStyles, makeStyles } from "@mui/styles";
+import Button from "@mui/material/Button";
+
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import MuiDialogTitle from "@mui/material/DialogTitle";
+import IconButton from "@mui/material/IconButton";
+import EditIcon from "@mui/icons-material/Edit";
+import Close from "@mui/icons-material/Close";
+import Typography from "@mui/material/Typography";
+import grey from "@mui/material/colors/grey";
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+
+import { showNotification } from "baselayer/components/Notifications";
+
+import * as SourceInGcnAction from "../ducks/sourcesingcn";
+
+dayjs.extend(utc);
+
+const useStyles = makeStyles((theme) => ({
+  container: {
+    width: "100%",
+    overflow: "scroll",
+  },
+  eventTags: {
+    marginLeft: "0.5rem",
+    "& > div": {
+      margin: "0.25rem",
+      color: "white",
+      background: theme.palette.primary.main,
+    },
+  },
+  buttons: {
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  button: {
+    maxWidth: "1.2rem",
+  },
+  buttonIcon: {
+    maxWidth: "1.2rem",
+  },
+}));
+const dialogTitleStyles = (theme) => ({
+  root: {
+    margin: 0,
+    padding: theme.spacing(2),
+  },
+  title: {
+    marginRight: theme.spacing(2),
+  },
+  closeButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: grey[500],
+  },
+});
+
+const DialogTitle = withStyles(dialogTitleStyles)(
+  ({ children, classes, onClose }) => (
+    <MuiDialogTitle className={classes.root}>
+      <Typography variant="h6" className={classes.title}>
+        {children}
+      </Typography>
+      {onClose ? (
+        <IconButton
+          aria-label="close"
+          className={classes.closeButton}
+          onClick={onClose}
+        >
+          <Close />
+        </IconButton>
+      ) : null}
+    </MuiDialogTitle>
+  )
+);
+
+const SourcesInGCN = ({
+  dateobs,
+  localizationName,
+  sourceId,
```

Comment:
I'm not sure what this component does. Does it give me a list of sources associated with the GCN or only checks if this source (given by `sourceId`) is included in the localization? Perhaps a short docstring is in place here. 
Also, `sourceId`->`sourceID` is preferred.

Reply:
the name was not clear at all. Will be fixed in the next commit. It is a component to confirm/reject/unconfirm a source from the source table when displayed on a gcn page.